test/unittest/compiler/unittest_interpreter.cpp

```diff
@@ -157,16 +157,28 @@ TEST_P(nntrainerInterpreterTest, graphSerializeAfterDeserialize) {
 }
 
 auto fc0 = LayerReprentation("fully_connected",
-                             {"name=fc0", "unit=1", "input_shape=1:1:100"});
+                             {"name=fc0", "unit=1", "input_shape=1:1:10"});
+auto fc1 = LayerReprentation("fully_connected",
+                             {"name=fc1", "unit=1", "input_shape=1:1:10"});
 
 auto flatten = LayerReprentation("flatten", {"name=flat"});
 
 #ifdef ENABLE_TFLITE_INTERPRETER
 TEST(flatbuffer, playground) {
+
+  auto manager = std::make_shared<nntrainer::Manager>();
+
   nntrainer::TfliteInterpreter interpreter;
-  auto g = makeGraph({fc0});
-  g->compile(nntrainer::LossType::LOSS_NONE);
+  auto g = makeGraph({fc0, fc1});
+  EXPECT_EQ(g->compile(nntrainer::LossType::LOSS_NONE), ML_ERROR_NONE);
+  EXPECT_EQ(g->initialize(manager), ML_ERROR_NONE);
+
+  manager->initializeWeights();
+  manager->allocateWeights();
+
   interpreter.serialize(g, "test.tflite");
+
+  manager->deallocateWeights();
```

Comment:
fyi) if i don't deallocateWeights here, there is a segfault. But I think it's gcc's inherent bug.
related link: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89303

Reply:
Yes, that is because destructor of manager does not call `deallocateWeights()`, `reset()` does which is called by `NeuralNetwork` destructor. So, if you don't call `deallocateWeights` or `reset` here, the freeing of memory happens strangely.

Why does manager does not call deallocate upon destruction? -> because manager must reset before graph resets as manager just holds the references to weight objects. If manager is updated to hold shared_ptr of weights than references, then constraint can be removed.

Reply:
Understood. I guess we should untangle a bit